web/components/Feed/Feed.tsx

```diff
@@ -299,18 +343,48 @@ function Feed(props: {
         } else {
           tempState[filt.key].push(filt.options[ind]);
         }
-        console.log(tempState);
-        return tempState;
+        toReturn = tempState;
+        break;
       default:
-        return state;
+        toReturn = state;
     }
+    setQueryFilters(getQueryFilters(toReturn));
+    return toReturn;
   }
 
+  const [queryFilters, setQueryFilters] = useState(
+    getQueryFilters(getInitialFilters())
+  );
+
   const [selectedFilters, setSelectedFilters] = useReducer(
     filterReducer,
     getInitialFilters()
   );
 
+  const feedPosts = trpc.post.getFilteredPosts
+    .useQuery(queryFilters)
+    ?.data?.map((p: IPost) => {
+      let firstImage = defaultDog.src;
+      for (let i = 0; i < p.attachments.length; i++) {
+        const a = p.attachments[i];
+        if (
+          a.toLowerCase().includes(".png") ||
+          a.toLowerCase().includes(".jpeg") ||
+          a.toLowerCase().includes(".jpg")
+        ) {
+          firstImage = a;
+          break;
+        }
+      }
+      return {
+        image: firstImage,
+        date: dayjs(p.date.toString()).format("MM/DD/YYYY hh:mm A").toString(),
```

Comment:
Pass in the date object directly (change the signature of `FeedPostCard`) and do the date parsing at that component's level